src/lists.rs

```diff
@@ -125,30 +163,156 @@ pub fn write_list<'b>(items: &[(String, String)], formatting: &ListFormatting<'b
                     line_len += 1;
                 }
 
-                line_len += item_width;
+                line_len += total_width;
             }
             _ => {}
         }
 
-        result.push_str(item);
+        // Pre-comments
+        if let Some(ref comment) = item.pre_comment {
+            result.push_str(&rewrite_comment(comment,
+                                             // Block style in non-vertical mode
+                                             tactic != ListTactic::Vertical,
+                                             1000,
```

Comment:
I'm not a fan of these magic 1000s - could you use an Option<usize> rather than a usize?